### PR TITLE
The uploader should retry when there's an error

### PR DESCRIPTION
**Pull request by mikehaney24**
Uploaders weren't always retrying in all the situations they should've been, so this hopefully addresses that. Also does a slight readability refactor.

### CODE REVIEW

File: GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTUploader.m [hunk 1]
```diff
@@ -202,51 +202,39 @@ - (void)uploadPackage:(GDTCORUploadPackage *)package {
...
-
-            default:
-              break;
-          }
         }
         pb_release(gdt_cct_LogResponse_fields, &logResponse);
       }
+      if (!futureUploadTime) {
+        GDTCORLogDebug("%@", @"CCT: The backend response failed to parse, so the next request "
+                             @"won't occur until 15 minutes from now");
```
Comment (morganchen12) [inline GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTUploader.m]: drive-by comment: looks like `decodingError` is not checked or logged after calling `GDTCCTDecodeLogResponse`, maybe log it here?

Comment (mikehaney24) [reply GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTUploader.m]: Done.

**Comment by mikehaney24** [comment]
> Is a test feasible?

Not super feasible for the retry change, but the other changes are well tested.
